packages/ckeditor5-ui/tests/bindings/addKeyboardHandlingForGrid.js

```diff
@@ -0,0 +1,274 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+import addKeyboardHandlingForGrid from '../../src/bindings/addKeyboardHandlingForGrid';
+import View from '../../src/view';
+import ButtonView from '../../src/button/buttonview';
+import { KeystrokeHandler, FocusTracker, Locale, keyCodes } from '@ckeditor/ckeditor5-utils';
+
+describe( 'addKeyboardHandlingForGrid()', () => {
+	let view, keystrokes, focusTracker, gridElementsCollection;
+
+	beforeEach( () => {
+		view = new TestView();
+		keystrokes = new KeystrokeHandler();
+		focusTracker = new FocusTracker();
+
+		view.render();
+
+		gridElementsCollection = view.createCollection();
+
+		for ( let i = 0; i < 7; i++ ) {
+			const button = new ButtonView( new Locale() );
+
+			button.render();
+			gridElementsCollection.add( button );
+			focusTracker.add( button.element );
+		}
+
+		addKeyboardHandlingForGrid( keystrokes, focusTracker, gridElementsCollection, 3 );
+
+		keystrokes.listenTo( view.element );
+	} );
+
+	afterEach( () => {
+		view.element.remove();
+		view.destroy();
+		keystrokes.destroy();
+		focusTracker.destroy();
+	} );
+
+	describe( 'basic arrow moves', () => {
+		it( 'arrowright moves focus to the next grid item', () => {
+			// before: [x][ ][ ]	after: [ ][x][ ]	key: →
+			//         [ ][ ][ ]	       [ ][ ][ ]
+			//         [ ]      	       [ ]
+			const keyEvtData = {
+				keyCode: keyCodes.arrowright,
+				preventDefault: sinon.spy(),
+				stopPropagation: sinon.spy()
+			};
```

Comment:
Tip: Move these together with `keystrokes.press( ... )` to helper functions at the end of the file named `pressLeftArrow()`, `pressRightArrow()`, etc.. Helps with redundancy and makes the code very easy to read in natural language.